radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java

```diff
@@ -825,6 +768,62 @@ public UnsignedAtom buildAtomWithFee(List<ParticleGroup> particleGroups) {
 		return new UnsignedAtom(new Atom(allParticleGroups, metaData));
 	}
 
+	/**
+	 * Represents an atomic transaction to be committed to the ledger
+	 */
+	public final class Transaction {
+		private final ArrayList<Action> actions = new ArrayList<>();
+		private Transaction() {
+		}
+
+		/**
+		 * Execute an action within this transaction
+		 * @param action the action to execute
+		 */
+		public void execute(Action action) {
+			this.actions.add(action);
+		}
+
+		/**
+		 * Commit the transaction onto the ledger
+		 * @return the results of committing
+		 */
+		public Result commit() {
+			return buildDisconnectedResult(actions.toArray(new Action[0])).connect();
+		}
+	}
+
+	/**
+	 * Create a new transaction
+	 *
+	 * @return a new transaction
+	 */
+	public Transaction transaction() {
+		return new Transaction();
+	}
+
+	private void stageActions(String uuid, List<Action> actions) {
```

Comment:
Can this be null here
`AtomStore.stageParticleGroup` should not accept null values, so this probably shouldn't either